### ISSUE TITLE
Inconsistent naming

### ISSUE THREAD

**Issue by monolithed** [nodeca/argparse#87]
```
{
     constant: [str | int | ...]
     action: [storeConst | appendConst | ...]
}
```

It should be 

```
{
     const: [String | Number | ...]
     action: [storeConst | appendConst | ...]
}
```

or

```
{
     const: [int | str | ...]
     action: [store_const | append_const | ...]
}
```

I see some contradictions like:
1. `const` is not reserved word as an object property
2. There'a no `int` and `str` types in JavaScript
3. Why camelCase (so why `--test-key` not transformed to `testKey`) ?

And so on...

Personally, I  prefer the last one

**Comment by puzrin** [comment]
It's too late to change API imho. Package is stable. I could consider some change, but only if someone decide to resync this port with latest argparse - that will be a reason to release 2.0.

About 'int' and so on... This is port, it should be as similar as possible. Also, some python types have no equivalent in js. `Number` is float in js, while `int` is integer in python.

**Comment by monolithed** [comment]
You are right that these changes can only be considered in release 2.0.

> Also, some python types have no equivalent in js. Number is float in js, while int is integer in python.
That's why I prefer the last example )

**Comment by puzrin** [comment]
Imho the best we can do now is to add note to differences list https://github.com/nodeca/argparse#argparse.

I'll keep this open, but honestly, no plans to resync port.

**Comment by puzrin** [comment]
Probably we could support both camelCase & snake_case.

**Comment by monolithed** [comment]
+1!

**Comment by rlidwka** [comment]
Fixed as part of argparse 2.0.

Use it now like this:

```
{
     const: ['int' | 'str' | Number | String | ...]
     action: ['store_const' | 'append_const' | ...]
}
```

Note that while `int` and `str` are functions in python, we support them as strings `'int'` and `'str'` for convenience.
